frontend/src/components/AuthButton.jsx

```diff
@@ -0,0 +1,219 @@
+import { useEffect, useState } from "react";
+
+const AuthButton = () => {
+  const [formData, setFormData] = useState({
+    email: "",
+    password: "",
+  });
+  const [loggedInStatus, setLoggedInStatus] = useState(false);
+  const [errorMsg, setErrorMsg] = useState("");
+
+  const checkLoginStatus = async () => {
+    try {
+      const response = await fetch("/authapi/profile", {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      setLoggedInStatus(response.ok); // Logged in only on 200 status
+    } catch (error) {
+      console.error("Error checking login status:", error);
+      setLoggedInStatus(false);
+    }
+  };
+  useEffect(() => {
+    checkLoginStatus();
+  }, []);
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((formValueState) => ({
+      ...formValueState,
+      [name]: value,
+    }));
+  };
+
+  const handleLogin = async (e) => {
+    if (e) {
+      e.preventDefault();
+    } else {
+      return;
+    }
+
+    try {
+      const response = await fetch("/authapi/login", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+
+      console.log("Login Status:", response.status);
+      console.log("Login Email:", formData.email);
+
+      // TODO: should we specify err here, or is it more secure to not give any
+      // info away? Does it matter if user doesn't exist, bad credentials etc.?
+      if (!response.ok) {
+        setErrorMsg("Login failed");
+        console.error("Login request rejected:", response.status);
+        return;
+      }
+
+      setLoggedInStatus(true);
+      document.getElementById("loginForm").close();
+      window.location.reload();
+    } catch (error) {
+      setErrorMsg("Login failed");
+      console.error("Login error:", error);
+    }
+  };
+
+  const handleRegister = async (e) => {
+    if (e) {
+      e.preventDefault();
+    } else {
+      return;
+    }
+    try {
+      const response = await fetch("/authapi/register", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+
+      console.log("Register status:", response.status);
+      console.log("Register email:", formData.email);
+      console.log("Register response:", response);
```

Comment:
We should replace this with a logging framework like the `logger` in the auth service. Even if we don't decide on one for a while, I'd rather remove these console.log statements or replace them with more useful messages ASAP.